src/gh/components/DF_remove_beam/code.py

```diff
@@ -0,0 +1,25 @@
+#! python3
+
+import typing
+
+from ghpythonlib.componentbase import executingcomponent as component
+
+import diffCheck
+import diffCheck.df_geometries
+import diffCheck.diffcheck_bindings
+import diffCheck.df_util
+
+
+class DFRemoveBeam(component):
+    def RunScript(self,
+            i_assembly : diffCheck.df_geometries.DFAssembly=None,
+            i_idx_2_remove : typing.List[int]=None):
+
+        if i_assembly is None or i_idx_2_remove is None:
+            return None
+
+        o_assembly = i_assembly.deepcopy()
```

Comment:
I wonder if this is not where the issue of the lag appears. That the deepcopy() is not 100% a deep copy and therefore some objects are shared between components... This is only an intuition and I will spend a bit more time on that tomorrow 28/8

Reply:
Might be a good hint acutally. Tell me if you find something fishy in the deepcopy() func!